File: app/src/main/java/farasense/mobile/view_model/YearlyChartConsumptionFragmentViewModel.java
package farasense.mobile.view_model;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import com.github.mikephil.charting.data.BarEntry;
import org.joda.time.Interval;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import farasense.mobile.model.DAO.FaraSenseSensorDailyDAO;
import farasense.mobile.model.realm.FaraSenseSensorDaily;
import farasense.mobile.util.DateUtil;

public class YearlyChartConsumptionFragmentViewModel extends AndroidViewModel {

    private List<String> yearlyChartLabels = new ArrayList<>();

    public YearlyChartConsumptionFragmentViewModel(Application application) { super(application); }

    public List<BarEntry> getYearlyConsumption() {
        int yearlyBehind = 0;
        List<BarEntry> entriesMeasures = new ArrayList<>();
        List<Double> measureList = new ArrayList<>();
        List<Interval> intervals10Year = DateUtil.getAllIntervalsLast10Year();

        do {
            List<FaraSenseSensorDaily> sensorMeasuresList = FaraSenseSensorDailyDAO.getMeasureByIntervals(
                    intervals10Year.get(yearlyBehind).getStart().toDate(),
                    intervals10Year.get(yearlyBehind).getEnd().toDate()
            );

            if (sensorMeasuresList != null) {
                Double totalKhm = 0.0;
                for (FaraSenseSensorDaily measure: sensorMeasuresList) {
                    totalKhm = totalKhm + measure.getTotalKwh();
                }

                measureList.add(totalKhm);
            } else {
                measureList.add(Double.valueOf(0));
            }

            yearlyChartLabels.add(String.valueOf(intervals10Year.get(yearlyBehind).getStart().getYear()));

            yearlyBehind++;
        } while (yearlyBehind < intervals10Year.size());

        Collections.reverse(measureList);

        int i = 0;
        for (Double measure : measureList) {
            entriesMeasures.add(i, new BarEntry(
                    Float.valueOf(i),
                    Float.valueOf(measure.floatValue())
            ));
            i++;
        }

        return entriesMeasures;
    }

    public List<String> getMonthlyChartLabels() {
        Collections.reverse(yearlyChartLabels);
        return yearlyChartLabels;
    }
}

File: app/src/main/java/farasense/mobile/view_model/YearlyChartConsumptionFragmentViewModel.kt
package farasense.mobile.view_model

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import com.github.mikephil.charting.data.BarEntry
import java.util.ArrayList
import farasense.mobile.model.DAO.FaraSenseSensorDailyDAO
import farasense.mobile.util.DateUtil

class YearlyChartConsumptionFragmentViewModel(application: Application) : AndroidViewModel(application) {

    private val yearlyChartLabels = ArrayList<String>()

    val yearlyConsumption: List<BarEntry>
        get() {
            var yearlyBehind = 0
            val entriesMeasures = ArrayList<BarEntry>()
            val measureList = ArrayList<Double>()
            val intervals10Year = DateUtil.allIntervalsLast10Year

            do {
                val sensorMeasuresList = FaraSenseSensorDailyDAO().getMeasureByIntervals(
                        intervals10Year[yearlyBehind].start.toDate(),
                        intervals10Year[yearlyBehind].end.toDate()
                )

                if (sensorMeasuresList != null) {
                    var totalKhm: Double? = 0.0
                    for (measure in sensorMeasuresList) {
                        totalKhm = totalKhm!! + measure.totalKwh!!
                    }

                    totalKhm?.let { measureList.add(it) }
                } else {
                    measureList.add(java.lang.Double.valueOf(0.0))
                }

                yearlyChartLabels.add(intervals10Year[yearlyBehind].start.year.toString())

                yearlyBehind++
            } while (yearlyBehind < intervals10Year.size)

            measureList.reverse()

            var i = 0
            for (measure in measureList) {
                entriesMeasures.add(i, BarEntry(
                        java.lang.Float.valueOf(i.toFloat()),
                        java.lang.Float.valueOf(measure.toFloat())
                ))
                i++
            }

            return entriesMeasures
        }

    val monthlyChartLabels: List<String>
        get() {
            yearlyChartLabels.reverse()
            return yearlyChartLabels
        }
}
